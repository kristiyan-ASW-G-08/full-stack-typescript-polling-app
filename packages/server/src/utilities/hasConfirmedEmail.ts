import RESTError, { errors } from '@utilities/RESTError';

const hasConfirmedEmail = (isConfirmed: boolean): void => {
  if (!isConfirmed) {
    const { status, message } = errors.Unauthorized;
    throw new RESTError(status, message, [
      {
        path: 'email',
        message: 'Confirm your email to proceed',
      },
    ]);
  }
};

export default hasConfirmedEmail;
