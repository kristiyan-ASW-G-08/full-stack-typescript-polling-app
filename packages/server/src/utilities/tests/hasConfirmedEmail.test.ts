import hasConfirmedEmail from '@utilities/hasConfirmedEmail';

describe('hasConfirmedEmail', (): void => {
  it('should throw an error when the passes value is truthy', async (): Promise<
    void
  > => {
    expect(() => hasConfirmedEmail(true)).not.toThrow();
  });
  it('should throw an error when the passed value is falsy', async (): Promise<
    void
  > => {
    expect(() => hasConfirmedEmail(false)).toThrowErrorMatchingSnapshot();
  });
});
