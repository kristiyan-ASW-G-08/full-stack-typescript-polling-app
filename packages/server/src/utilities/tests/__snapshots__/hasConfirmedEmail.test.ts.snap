// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hasConfirmedEmail should throw an error when the passed value is falsy 1`] = `"Authentication credentials not valid or missing"`;
